agent-connection/src/main/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/PythonRefAgentConnection.kt

```diff
@@ -0,0 +1,233 @@
+package com.luxoft.blockchainlab.corda.hyperledger.indy
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.luxoft.blockchainlab.hyperledger.indy.*
+import com.luxoft.blockchainlab.hyperledger.indy.utils.SerializationUtils
+import mu.KotlinLogging
+import org.java_websocket.client.WebSocketClient
+import org.java_websocket.handshake.ServerHandshake
+import java.lang.Thread.sleep
+import java.net.URI
+import java.time.Instant
+import java.util.*
+import kotlin.RuntimeException
+
+
+class PythonRefAgentConnection : AgentConnection {
+    override fun disconnect() {
+        if (getConnectionStatus() == AgentConnectionStatus.AGENT_CONNECTION_CONNECTED) {
+            webSocket.closeBlocking()
```

Comment:
Non blocking mode will require to have current state handling on every request. This will add a lot of complexity. 
IMHO, it`s essential for our use cases to wait for connection to actually open\close, to run other logic.